### PR TITLE
Fix exceptions in IntervalCalculation and ResultIndexingHandler

### PR DESCRIPTION
**Pull request by kaituo**
### Description
- **IntervalCalculation**: Prevent an `ArrayIndexOutOfBoundsException` by returning early when there are fewer than two timestamps. Previously, the code assumed at least two timestamps, causing an exception when only one was present.

- **ResultIndexingHandler**: Handle exceptions from asynchronous calls by logging error messages instead of throwing exceptions. Since the caller does not wait for these asynchronous operations, throwing exceptions had no effect and could lead to unhandled exceptions. Logging provides visibility without disrupting the caller's flow.

Testing done:
1. added UT and ITs.

Signed-off-by: Kaituo Li <kaituo@amazon.com>

### Check List
- [X] New functionality includes testing.
- [X] New functionality has been documented.
- [X] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [X] Commits are signed per the DCO using `--signoff`.
- [X] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/anomaly-detection/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by owaiskazi19 — COMMENTED** [review]
Took an initial pass

File: src/main/java/org/opensearch/forecast/transport/ForecastResultBulkTransportAction.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public ForecastResultBulkTransportAction(
     }
 
     @Override
-    protected BulkRequest prepareBulkRequest(float indexingPressurePercent, ForecastResultBulkRequest request) {
+    public BulkRequest prepareBulkRequest(float indexingPressurePercent, ForecastResultBulkRequest request) {
```
[the remark this answers is not in the document]
Comment (kaituo) [reply src/main/java/org/opensearch/forecast/transport/ForecastResultBulkTransportAction.java:61]: package private cannot be seen in different packages. The super type method is overridden by two different classes in different packages Thus, we cannot use package private.

File: src/main/java/org/opensearch/timeseries/rest/handler/IntervalCalculation.java [hunk 2]
```diff
@@ -252,8 +252,9 @@ private void findMinimumInterval(LongBounds timeStampBounds, ActionListener<Inte
                 .createSearchRequest(new IntervalTimeConfiguration(1, ChronoUnit.MINUTES), timeStampBounds, topEntity);
             final ActionListener<SearchResponse> searchResponseListener = ActionListener.wrap(response -> {
                 List<Long> timestamps = aggregationPrep.getTimestamps(response);
-                if (timestamps.isEmpty()) {
-                    logger.warn("empty data, return one minute by default");
+                if (timestamps.size() < 2) {
+                    // to calculate the difference we need at least 2 timestamps
+                    logger.warn("not enough data, return one minute by default");
```
[the remark this answers is not in the document]
Comment (kaituo) [reply src/main/java/org/opensearch/timeseries/rest/handler/IntervalCalculation.java:257]: Both are ok. I prefer the old one as "one minute" is more explicit, as "one" emphasizes the specific duration of one minute. It makes it clear that exactly one minute is being returned. "A minute" sounds a bit more casual and could imply any single minute without the same level of emphasis as "one."

**Review by owaiskazi19 — APPROVED** [review]
LG

File: src/main/java/org/opensearch/forecast/transport/ForecastResultBulkTransportAction.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (owaiskazi19) [reply src/main/java/org/opensearch/forecast/transport/ForecastResultBulkTransportAction.java:61]: Ah! I see different packages. Makes sense

**Pull request merged**